homeassistant/components/glances/__init__.py

```diff
@@ -27,6 +28,16 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
 
     await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)
 
+    if config_entry.data[CONF_VERSION] == 2:
```

Comment:
Is the version auto updated?

Reply:
We must allow the user to solve the issue when creating the issue. Currently the only way is to remove the config entry and set up a new one. That's not good. We should allow the user to change the version of the existing config entry.

Reply:
Maybe the version is good to put in an options flow?

Reply:
If we assume that the user will upgrade to v3 of the glances server while keeping the same host and credentials then it should be fine to change the version key in the entry data to 3. But I don't think its good to have it as an option as an entry with v3 will have some sensors that are not available in v2. 
Here is an example: https://github.com/home-assistant/core/issues/99735

Reply:
We can validate that the server is compatible with the selected version before updating, eg by trying to fetch a key that is only available in version 3.

It would be even better if the user didn't have to select version and we just automatically determine what version is supported by the running server during config entry setup. Can we try version 3, then version 2, then error out? If only version 2 is available we create the repair issue and ask the user to reload the config entry after upgrading the server.

Reply:
The url endpoint for connecting to the api includes [the version](https://github.com/home-assistant-ecosystem/python-glances-api/blob/1cadd3ce5f8a1012c00a1d5727550a9900ce985b/glances_api/__init__.py#L35). I can try version 3 then version 2 in case version 3 endpoint gives an error.
We still need to address issue #99735 by updating glances api library itself to handle keys that are not available in v2. I will submit a PR for this.